plenum/common/stack_manager.py

```diff
@@ -76,35 +77,45 @@ def parseLedgerForHaAndKeys(ledger, returnActive=True):
-                clientStackName = nodeName + CLIENT_STACK_SUFFIX
-                nHa = (txn[DATA][NODE_IP], txn[DATA][NODE_PORT]) \
-                    if (NODE_IP in txn[DATA] and NODE_PORT in txn[DATA]) \
-                    else None
-                cHa = (txn[DATA][CLIENT_IP], txn[DATA][CLIENT_PORT]) \
-                    if (CLIENT_IP in txn[DATA] and CLIENT_PORT in txn[DATA]) \
-                    else None
-                if nHa:
-                    nodeReg[nodeName] = HA(*nHa)
-                if cHa:
-                    cliNodeReg[clientStackName] = HA(*cHa)
-
-                try:
-                    # TODO: Need to handle abbreviated verkey
-                    verkey = cryptonymToHex(txn[TARGET_NYM])
-                except ValueError as ex:
-                    raise ValueError("Invalid verkey. Rebuild pool transactions.")
-
-                nodeKeys[nodeName] = verkey
-
-                services = txn[DATA].get(SERVICES)
-                if isinstance(services, list):
-                    if VALIDATOR in services:
-                        activeValidators.add(nodeName)
-                    else:
-                        activeValidators.discard(nodeName)
+        try:
+            for _, txn in ledger.getAllTxn():
+                if txn[TXN_TYPE] == NODE:
+                    nodeName = txn[DATA][ALIAS]
+                    clientStackName = nodeName + CLIENT_STACK_SUFFIX
+                    nHa = (txn[DATA][NODE_IP], txn[DATA][NODE_PORT]) \
+                        if (NODE_IP in txn[DATA] and NODE_PORT in txn[DATA]) \
+                        else None
+                    cHa = (txn[DATA][CLIENT_IP], txn[DATA][CLIENT_PORT]) \
+                        if (CLIENT_IP in txn[DATA] and CLIENT_PORT in txn[DATA]) \
+                        else None
+                    if nHa:
+                        nodeReg[nodeName] = HA(*nHa)
+                    if cHa:
+                        cliNodeReg[clientStackName] = HA(*cHa)
+
+                    try:
+                        # TODO: Need to handle abbreviated verkey
+                        key_type = 'verkey'
+                        verkey = cryptonymToHex(txn[TARGET_NYM])
+                        key_type = 'identifier'
+                        cryptonymToHex(txn[IDENTIFIER])
+                    except ValueError as ex:
+                        logger.debug('Invalid {}. Rebuild pool transactions.'.format(key_type))
+                        print('Invalid {}. Rebuild pool transactions.'.format(key_type))
+                        exit(-1)
+
+                    nodeKeys[nodeName] = verkey
+
+                    services = txn[DATA].get(SERVICES)
+                    if isinstance(services, list):
+                        if VALIDATOR in services:
+                            activeValidators.add(nodeName)
+                        else:
+                            activeValidators.discard(nodeName)
+        except JSONDecodeError as exc:
```

Comment:
Yes, I can find a more generic exception for that and yes I will merge it into M1.